tests/unittests/Foundation/NSFileManagerTests.mm

```diff
@@ -218,3 +181,127 @@
+
+    NSDictionary* retrievedAttributes = [manager attributesOfItemAtPath:[srcURL path] error:&error];
+    EXPECT_OBJCEQ(NSFileTypeRegular, retrievedAttributes[NSFileType]);
+    EXPECT_OBJCEQ(@NO, retrievedAttributes[NSFileImmutable]);
+    EXPECT_OBJCEQ(@20, retrievedAttributes[NSFileSize]);
+
+    NSDate* modificationDate = retrievedAttributes[NSFileModificationDate];
+    EXPECT_NE(nil, modificationDate);
+    EXPECT_NEAR([rightNow timeIntervalSince1970], [modificationDate timeIntervalSince1970], 10);
+
+    NSDate* creationDate = retrievedAttributes[NSFileCreationDate];
+    EXPECT_NE(nil, creationDate);
+    EXPECT_NEAR([rightNow timeIntervalSince1970], [creationDate timeIntervalSince1970], 10);
+
+    EXPECT_EQ(nil, error);
+}
+
+TEST(NSFileManager, EnumeratorAtURL) {
+    NSString* subFolderName = @"SUBFOLDER";
+    NSString* itemName = @"ArBiTraRY_FiLE.txt";
+
+    NSFileManager* manager = [NSFileManager defaultManager];
+    [manager createDirectoryAtPath:subFolderName withIntermediateDirectories:NO attributes:nil error:nullptr];
+    createFileWithContentAndVerify(itemName, @"The Quick Brown Fox.");
+    SCOPE_DELETE_FILE(subFolderName);
+    SCOPE_DELETE_FILE(itemName);
+    NSString* currentDirectory = [manager currentDirectoryPath];
+
+    NSURL* current = [NSURL fileURLWithPath:currentDirectory];
+    NSDirectoryEnumerator* enumerator = [manager enumeratorAtURL:current
+                                      includingPropertiesForKeys:nil
+                                                         options:NSDirectoryEnumerationSkipsHiddenFiles
+                                                    errorHandler:^(NSURL* url, NSError* error) {
+                                                        return NO;
+                                                    }];
+
+    NSArray* all = [enumerator allObjects];
+    EXPECT_TRUE([all containsObject:[NSURL fileURLWithPath:[currentDirectory stringByAppendingPathComponent:subFolderName]]]);
+    EXPECT_TRUE([all containsObject:[NSURL fileURLWithPath:[currentDirectory stringByAppendingPathComponent:itemName]]]);
+}
+
+TEST(NSFileManager, SubPathsOfDirectoryAtPath) {
+    NSString* subFolderName = @"SUBFOLDER";
+    NSString* subSubFolderName = @"SUBFOLDER/SUBSUBFOLDER";
+    NSFileManager* manager = [NSFileManager defaultManager];
+    [manager createDirectoryAtPath:subFolderName withIntermediateDirectories:NO attributes:nil error:nullptr];
+    [manager createDirectoryAtPath:subSubFolderName withIntermediateDirectories:NO attributes:nil error:nullptr];
+    SCOPE_DELETE_FILE(subFolderName);
+    SCOPE_DELETE_FILE(subSubFolderName);
+
+    NSArray* subpaths = [manager subpathsOfDirectoryAtPath:[manager currentDirectoryPath] error:nullptr];
+
+    EXPECT_TRUE([subpaths containsObject:subFolderName]);
+    EXPECT_TRUE([subpaths containsObject:subSubFolderName]);
+
+    EXPECT_FALSE([subpaths containsObject:@"."]);
+    EXPECT_FALSE([subpaths containsObject:@".."]);
+}
```

Comment:
can you add some equivalent tests for files/folders that don't exist and make sure everything's handled properly?